### ISSUE TITLE
Make CallCredentials non-experimental

### ISSUE THREAD

**Issue by zhangkun83** [grpc/grpc-java#1914]
This is a tracking issue for removing `@ExperimentalApi` from `CallCredentials`-related API.

**Comment by dapengzhang0** [comment]
not stabilized yet, keep unscheduled

**Comment by garrettjonesgoogle** [comment]
This seems critical for client stability - without this being non-experimental, there is no non-experimental way to make gRPC calls that require credentials, effectively making it so that gRPC doesn't have a "complete" GA surface.

**Comment by ejona86** [comment]
Agreed, this is very important to stabilize. If push comes to shove, there is one trick we could do: we could stabilize the usages, but not the implementations. So that would mean CallOptions.withCallCredentials and MoreCallCredentials could be stable, even though the interface may change. We'd basically say that the name would remain the same and MoreCallCredentials would remain compatible. grpc-auth would need to be version-pinned (if it isn't already) to the same version of grpc-core, but that shouldn't cause much issue.

**Comment by ejona86** [comment]
I'm assigning this to myself to remove most of the experimental api tags as I mentioned in the last comment, but I'm not going to end up stabilizing all of the API.

**Comment by saturnism** [comment]
Would it be possible not to have Google credentials in the core gRPC libraries? gRPC as a core RPC framework probably don't need Google credentials.

**Comment by ejona86** [comment]
@saturnism, only `grpc-auth` depends on `google-auth-library-credentials`. Also, "Google credentials" isn't in `google-auth-library-credentials`, but in `google-auth-library-oauth2-http`. `CallCredentials` is in `grpc-core`, so there should be no unnecessary dependencies.

**Comment by saturnism** [comment]
it seems grpc-auth is only dealing w/ google credentials. @garrettjonesgoogle do you actually need this from gcloud-java? It was a discussion point of dependency conflicts.

**Comment by ejona86** [comment]
@saturnism, your comments are off-topic for this issue (CallCredentials is in grpc-core and has no external dependencies). Let's move the conversation to the freshly-created #3280

**Comment by ejona86** [comment]
Possibly need to fix: there is no way to cancel an async CallCredential. C has a [cancel in its API](https://github.com/grpc/grpc/blob/v1.8.0/src/core/lib/security/credentials/credentials.h#L174).

**Comment by ejona86** [comment]
Hmm... maybe that is an internal API. Because I also see [this API](https://github.com/grpc/grpc/blob/v1.8.0/include/grpc/grpc_security.h#L261) which doesn't have a cancel.

**Comment by ejona86** [comment]
API review:
 * Ready to stabilize
 * Obviously will remove the thisUsesUnstableApi() method. Probably in two phases: give it an implementation and make it deprecated, later delete.

**Comment by temawi** [comment]
Stabilized in #10208.

The `thisUsesUnstableApi()` method has been marked deprecated to give implementers time to stop using it until we remove the method in the future. I will leave this issue open to track the removal of the method.
